src/cunumeric/index/putmask_template.inl

```diff
@@ -52,9 +52,6 @@ struct Putmask {
   Putmask(PutmaskArgs& args) : dense(false)
   {
     rect = args.input.shape<DIM>();
-#ifdef DEBUG_CUNUMERIC
-    assert(rect == args.mask.shape<DIM>());
-#endif
```

Comment:
Would this still be useful to have? In addition to a size check, it also verifies that tile boundaries align on the point tasks (e.g. sub-region `(4,4)-(4,7)` on `input` is matched to sub-region `(4,4)-(4,7)` on `mask`).

Reply:
since the key (mask) can be broadcasted in some cases, this assertion not always true (I believe shape<DIM>() will return not a correct shape for the broadcasted mask)

Reply:
But to check this case I added `assert self.shape == mask.shape` on the python side for `def putmask`